crates/bevy_sprite/src/picking_backend.rs

```diff
@@ -135,12 +168,44 @@ pub fn sprite_picking(
 
                 let is_cursor_in_sprite = rect.contains(cursor_pos_sprite);
 
-                blocked = is_cursor_in_sprite
+                let cursor_in_valid_pixels_of_sprite = match settings.picking_mode {
+                    SpritePickingMode::AlphaThreshold(cutoff) if is_cursor_in_sprite => {
+                        let texture: &Image = images.get(&sprite.image)?;
+                        // If using a texture atlas, grab the offset of the current sprite index. (0,0) otherwise
+                        let texture_rect = sprite
+                            .texture_atlas
+                            .as_ref()
+                            .and_then(|atlas| {
+                                texture_atlas_layout
+                                    .get(&atlas.layout)
+                                    .map(|f| f.textures[atlas.index])
+                            })
+                            .or(Some(URect::new(0, 0, texture.width(), texture.height())))?;
+                        // get mouse position on texture
+                        let texture_position = (texture_rect.center().as_vec2()
+                            + cursor_pos_sprite.trunc())
+                        .as_uvec2();
+                        // grab pixel
+                        let pixel_index =
+                            (texture_position.y * texture.width() + texture_position.x) as usize;
+                        // check transparency
+                        match texture.data.get(pixel_index * 4..(pixel_index * 4 + 4)) {
+                            // If possible check the alpha bit is above cutoff
+                            Some(pixel_data) if pixel_data[3] > cutoff => true,
```

Comment:
Can we move the condition to replace `true`? If there is pixel data the result is just the alpha cutoff expression.